ReSharper.FSharp/src/FSharp.Psi.Intentions/src/QuickFixes/UseNestedRecordFieldSyntaxFix.fs

```diff
@@ -0,0 +1,45 @@
+namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.QuickFixes
+
+open System.Collections.Generic
+open JetBrains.ReSharper.Plugins.FSharp.Psi
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.Highlightings
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Refactorings
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Impl
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Impl.Tree
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Tree
+open JetBrains.ReSharper.Psi.ExtensionsAPI
+open JetBrains.ReSharper.Resources.Shell
+
+type UseNestedRecordFieldSyntaxFix(warning: NestedWithCanBeSimplifiedWarning) =
+    inherit FSharpScopedQuickFixBase(warning.Binding)
+
+    let bindingExpr = warning.Binding
+
+    let rec getFieldToUpdateData (expr: IFSharpExpression) (currentFieldPath: List<string>) =
+        match expr.IgnoreInnerParens() with
+        | :? IRecordExpr as recordExpr ->
+            let field = recordExpr.FieldBindings.SingleItem
```

Comment:
> we can easily add a test that will enforce this contract between the analyzer and the quick fix

Yes, if there were several elements

> Should we be more defensive in the quick fix, just in case the analyzer logic changes in future?

for sure